nncf/torch/quantization/layers.py

```diff
@@ -1094,7 +1094,7 @@ def __init__(self, scale: torch.Tensor, zero_point: torch.Tensor, result_dtype:
         """
         super().__init__()
         self.register_buffer("_scale", scale.type(dtype=torch.float16))
-        self.register_buffer("_zero_point", self.pack_weight(zero_point))
+        self.register_buffer("_zero_point", self.pack_weight(zero_point.type(dtype=torch.uint8)))
```

Comment:
@alexsu52 please confirm the fix

Reply:
Motivation is that `zero_point` is used in ScaleEstimation quantization formula as float type, not unit type. See these:
https://github.com/openvinotoolkit/nncf/blob/develop/nncf/quantization/algorithms/weight_compression/scale_estimation.py#L226
https://github.com/openvinotoolkit/nncf/blob/develop/nncf/quantization/algorithms/weight_compression/openvino_backend.py#L377
This type conversion is hidden for OV because it applies type transformation from float to dtype in the transformation directly - https://github.com/openvinotoolkit/nncf/blob/develop/nncf/quantization/algorithms/weight_compression/openvino_backend.py#L252
So, I basically did the same for Torch and transform zero_point to expected dtype in model transformation step.